server/system/service/smtp_connection_checker.go

```diff
@@ -0,0 +1,159 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"github.com/cortezaproject/corteza/server/pkg/actionlog"
+	intAuth "github.com/cortezaproject/corteza/server/pkg/auth"
+	"github.com/cortezaproject/corteza/server/pkg/mail"
+	"github.com/cortezaproject/corteza/server/pkg/options"
+	"github.com/cortezaproject/corteza/server/system/types"
+	gomail "gopkg.in/mail.v2"
+	htpl "html/template"
+	"io/ioutil"
+	"strings"
+)
+
+type (
+	smtpConnectionChecker struct {
+		actionlog actionlog.Recorder
+		settings  *types.AppSettings
+		ts        TemplateService
+		opt       options.AuthOpt
+	}
+)
+
+func SmtpConnectionChecker(s *types.AppSettings, ts TemplateService, opt options.AuthOpt) *smtpConnectionChecker {
+	return &smtpConnectionChecker{
+		actionlog: DefaultActionlog,
+		settings:  s,
+		ts:        ts,
+		opt:       opt,
+	}
+}
+
+// SmtpServerChecker Expected to error out
+func (svc smtpConnectionChecker) SmtpServerChecker(ctx context.Context, smtpConfigs *types.SmtpServer) (err error) {
```

Comment:
Let us restructure this function a bit.

The desired goal is not to send an email but to check the SMTP configuration.
So, invalid SMTP settings, and inability to send email is not an error but a result of this check (you expect it that it might not work).
But the invalid input parameters or inability to load the template you use to send an check email is an actual error (you are expecting it to be valid and to work).

So, return 2 vars; the first one would be a struct with checks that passed or failed (and reasons for them). Could be a simple:

```go
SmtpCheckResult struct {
   Host error `json:"host"`
   Port error
   ServerCheck error
   SendCheck error
  // ....?
   Valid bool `json:....`
}
```